server/py/framework/db/sqldb/db.py

```diff
@@ -571,6 +576,13 @@ def _enrich_run_model(self, now: datetime, run: Run, run_data: dict):
         ):
             end_time = run_end_time(run_data)
             self._update_run_end_time(run, run_data, now=end_time)
+        elif (
```

Comment:
looks code dup with above
```
f (
            run.state in mlrun.common.runtimes.constants.RunStates.terminal_states()
            and not run.end_time
        ):
            self._update_run_end_time(run, struct, now=datetime.now(timezone.utc))
        elif (
            run.state not in mlrun.common.runtimes.constants.RunStates.terminal_states()
        ):
            # Ensure end time is not set if the run is not in a terminal state
            run.end_time = None
            struct.setdefault("status", {}).pop("end_time", None)
```


can be moved out ?

Reply:
It's similar but I kept this `end_time = run_end_time(run_data)` for "store run" because I think it might be useful to allow setting the end time by client input via store.
Can extract it with `end_time` parameter.